tests/integration/test_sqs.py

```diff
@@ -1398,16 +1400,631 @@ def test_batch_send_with_invalid_char_should_succeed(self, sqs_client, sqs_creat
+
+        attributes = {
+            "attr1": {"StringValue": "test1", "DataType": "String"},
+            "attr2": {"StringValue": "test2", "DataType": "String"},
+        }
+        result_send = sqs_client.send_message(
+            QueueUrl=queue_url, MessageBody="test", MessageAttributes=attributes
+        )
+
+        result_receive = sqs_client.receive_message(
+            QueueUrl=queue_url, MessageAttributeNames=["All"]
+        )
+        messages = result_receive["Messages"]
+
+        assert messages[0]["MessageId"] == result_send["MessageId"]
+        assert messages[0]["MessageAttributes"] == attributes
+        assert messages[0]["MD5OfMessageAttributes"] == result_send["MD5OfMessageAttributes"]
+
+    def test_sent_message_retains_attributes_after_receive(self, sqs_client, sqs_create_queue):
+        # Old name: test_send_message_retains_attributes
+        queue_name = "queue-{}".format(short_uid())
+        queue_url = sqs_create_queue(QueueName=queue_name)
+
+        attributes = {"attr1": {"StringValue": "test1", "DataType": "String"}}
+        sqs_client.send_message(
+            QueueUrl=queue_url, MessageBody="test", MessageAttributes=attributes
+        )
+
+        # receive should not interfere with message attributes
+        sqs_client.receive_message(
+            QueueUrl=queue_url, VisibilityTimeout=0, MessageAttributeNames=["All"]
+        )
+        receive_result = sqs_client.receive_message(
+            QueueUrl=queue_url, MessageAttributeNames=["All"]
+        )
+        assert receive_result["Messages"][0]["MessageAttributes"] == attributes
+
+    def test_send_message_with_invalid_string_attributes(self, sqs_client, sqs_create_queue):
+        queue_name = "queue-{}".format(short_uid())
+        queue_url = sqs_create_queue(QueueName=queue_name)
+
+        # String Attributes must not contain non-printable characters
+        # See: https://docs.aws.amazon.com/AWSSimpleQueueService/latest/APIReference/API_SendMessage.html
+        invalid_attribute = {
+            "attr1": {"StringValue": "Invalid-{},{}".format(chr(8), chr(11)), "DataType": "String"}
+        }
+
+        with pytest.raises(Exception) as e:
+            sqs_client.send_message(
+                QueueUrl=queue_url, MessageBody="test", MessageAttributes=invalid_attribute
+            )
+        e.match("Invalid")
+
+    def test_send_message_with_invalid_payload_characters(self, sqs_client, sqs_create_queue):
+        queue_name = "queue-{}".format(short_uid())
+        queue_url = sqs_create_queue(QueueName=queue_name)
+        invalid_message_body = "Invalid-{}-{}-{}-{}".format(chr(0), chr(8), chr(19), chr(65535))
+
+        with pytest.raises(Exception) as e:
+            sqs_client.send_message(QueueUrl=queue_url, MessageBody=invalid_message_body)
+        e.match("InvalidMessageContents")
+
+    def test_dead_letter_queue_config(self, sqs_client, sqs_create_queue):
+        # TODO: not tested against AWS
```

Comment:
anything blocking the testing against AWS?

Reply:
No, was a leftover